gorm-tools/src/main/groovy/gorm/tools/databinding/EntityMapBinder.groovy

```diff
@@ -348,12 +354,14 @@ class EntityMapBinder extends SimpleDataBinder implements MapBinder {
             String sval = value as String
             //if its an empty string then its nothing so return
             if(!sval.trim()) return
-            //convert to long
-            value = sval as Long
+            //check id type, and convert to long/uuid based on type
+            Class idType = getIdPropertyType(association.getType())
+            if(Number.isAssignableFrom(idType)) { value = sval as Long }
+            else if (UUID.isAssignableFrom(idType)) { value = UUID.fromString(sval) }
```

Comment:
where are you getting these single line if/else syntax? I am not aware of any coding standards that would allow these in javascript lint or java. is there one?
also, on above, what if its a string, dont we have that in couple spots? this skips over anything that is not number or UUID. but I guess those can be exceptions as didnt deal with them in the change below where we used to specifically check for /number.